library/ssl_tls12_client.c

```diff
@@ -3360,7 +3360,7 @@ static int ssl_write_certificate_verify( mbedtls_ssl_context *ssl )
     const mbedtls_ssl_ciphersuite_t *ciphersuite_info =
         ssl->handshake->ciphersuite_info;
     size_t n = 0, offset = 0;
-    unsigned char hash[48];
+    unsigned char hash[PSA_HASH_MAX_SIZE];
```

Comment:
So what happens if we have `MBEDTLS_USE_PSA_CRYPTO` undefined, `PSA_WANT_ALG_SHA_512` and `PSA_WANT_ALG_SHA_384` both undefined, but `MBEDTLS_SHA512_C` defined?

A bit further up, around line 2391, we have:

```
#if defined(MBEDTLS_USE_PSA_CRYPTO)
    unsigned char hash[PSA_HASH_MAX_SIZE];
#else
    unsigned char hash[MBEDTLS_MD_MAX_SIZE];
#endif
```

Reply:
My (default) configuration has `MBEDTLS_USE_PSA_CRYPTO` undefined, and the library builds just fine.
You are raising a good point about interaction of the PSA and legacy modules in this aspect.